### PR TITLE
riotboot/flashwrite: support non-uniform flash page sizes

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

This removes all assumptions on uniform flash page sizes in the `riotboot_flashwrite` code when the non-pagewise mode is used. This is required to support riotboot on the stm32f{2,4,7}.

### Testing procedure

The automated test with `examples/suit_update` must still work.

### Issues/PRs references

Depends on #15412

### CODE REVIEW

**Comment by bergzand** [comment]
Rebased!

**Comment by aabadie** [comment]
#15412 is merge, please rebase

**Comment by bergzand** [comment]
Rebased!

File: sys/riotboot/flashwrite.c [hunk 1]
```diff
@@ -97,13 +97,15 @@ int riotboot_flashwrite_putbytes(riotboot_flashwrite_t *state,
...
 
     while (len) {
-        size_t flashpage_pos = state->offset % FLASHPAGE_SIZE;
         size_t flashwrite_buffer_pos = state->offset % RIOTBOOT_FLASHPAGE_BUFFER_SIZE;
         size_t flashpage_avail = RIOTBOOT_FLASHPAGE_BUFFER_SIZE - flashwrite_buffer_pos;
 
         size_t to_copy = min(flashpage_avail, len);
 
-        if (CONFIG_RIOTBOOT_FLASHWRITE_RAW && flashpage_pos == 0) {
+        void * target_addr = (uint8_t*)riotboot_slot_get_hdr(state->target_slot) + state->offset;
```
Comment (aabadie) [inline sys/riotboot/flashwrite.c:105]:
```suggestion
        void *target_addr = (uint8_t *)riotboot_slot_get_hdr(state->target_slot) + state->offset;
```

Comment (aabadie) [reply sys/riotboot/flashwrite.c:105]: squash this directly!

**Comment by fjmolinas** [comment]
ping @bergzand !

**Comment by benpicco** [comment]
So is the verdict to clean this up later together with the rest of the file?

File: sys/riotboot/flashwrite.c [hunk 2]
```diff
@@ -56,6 +54,8 @@ int riotboot_flashwrite_init_raw(riotboot_flashwrite_t *state, int target_slot,
     state->target_slot = target_slot;
     state->flashpage = flashpage_page((void *)riotboot_slot_get_hdr(target_slot));
 
+    assert(offset <= flashpage_size(state->flashpage));
```
Comment (benpicco) [inline sys/riotboot/flashwrite.c:57]: `flashpage_size()` does not exist yet

**Comment by benpicco** [comment]
Is this still needed?

**Comment by bergzand** [comment]
> Is this still needed?

I think I accidentally skipped over this one :(

**Pull request closed without merging**